server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java

```diff
@@ -77,19 +77,23 @@ public void read(DbFileSources.Line.Builder lineBuilder) {
 
   private void processSymbols(DbFileSources.Line.Builder lineBuilder) {
     int line = lineBuilder.getLine();
-    List<BatchReport.Symbol> lineSymbols = findSymbolsMatchingLine(line);
+
+    List<BatchReport.Symbol> lineSymbols = new ArrayList<>(this.symbolsPerLine.get(line));
+    // Sort symbols to have deterministic results and avoid false variation that would lead to an unnecessary update of the source files
+    // data
+    Collections.sort(lineSymbols, SymbolsComparator.INSTANCE);
```

Comment:
I need a HashSetMultimap to remove duplicate symbols per line based on their hash
a SortedSetMultimap will use the comparator to both sort and remove duplicates, and it will use it for each insert. The current comparator implementation is not strong enough to be used for comparison.
in addition, hash method of protobuf objects is memoized for better performance
I'd rather sort once and at the _only_ time the list of symbols of a line is used